src/app.ts

```diff
@@ -22,108 +22,112 @@ import {voteOnSponsorTime} from './routes/voteOnSponsorTime';
 import {getSkipSegmentsByHash} from './routes/getSkipSegmentsByHash';
 import {postSkipSegments} from './routes/postSkipSegments';
 import {endpoint as getSkipSegments} from './routes/getSkipSegments';
-
 import {userCounter} from './middleware/userCounter';
 import {loggerMiddleware} from './middleware/logger';
 import {corsMiddleware} from './middleware/cors';
 import {rateLimitMiddleware} from './middleware/requestRateLimit';
 
-// Create a service (the app object is just a callback).
-const app = express();
-// Rate limit endpoint lists
-const voteEndpoints: RequestHandler[] = [voteOnSponsorTime];
-const viewEndpoints: RequestHandler[] = [viewedVideoSponsorTime];
-if (config.rateLimit) {
-    if (config.rateLimit.vote) voteEndpoints.unshift(rateLimitMiddleware(config.rateLimit.vote));
-    if (config.rateLimit.view) viewEndpoints.unshift(rateLimitMiddleware(config.rateLimit.view));
-}
 
-//setup CORS correctly
-app.use(corsMiddleware);
-app.use(loggerMiddleware);
-app.use(express.json());
+export function createServer(callback: () => void) {
+    // Create a service (the app object is just a callback).
+    const app = express();
+
+    //setup CORS correctly
+    app.use(corsMiddleware);
+    app.use(loggerMiddleware);
+    app.use(express.json());
+
+    if (config.userCounterURL) app.use(userCounter);
 
-if (config.userCounterURL) app.use(userCounter);
+    // Setup pretty JSON
+    if (config.mode === "development") app.set('json spaces', 2);
 
-// Setup pretty JSON
-if (config.mode === "development") app.set('json spaces', 2);
+    // Set production mode
+    app.set('env', config.mode || 'production');
 
-// Set production mode
-app.set('env', config.mode || 'production');
+    setupRoutes(app);
 
-//add the get function
-app.get('/api/getVideoSponsorTimes', oldGetVideoSponsorTimes);
+    return app.listen(config.port, callback);
+}
 
-//add the oldpost function
-app.get('/api/postVideoSponsorTimes', oldSubmitSponsorTimes);
-app.post('/api/postVideoSponsorTimes', oldSubmitSponsorTimes);
+function setupRoutes(app: Express) {
```

Comment:
need to mock middleware before it's actually used, so extracted this into a separate function